modules/oe_list_pages_open_vocabularies/src/EventSubscriber/OpenVocabulariesConfigSubscriber.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Drupal\oe_list_pages_open_vocabularies\EventSubscriber;
+
+use Drupal\Core\Config\ConfigCrudEvent;
+use Drupal\Core\Config\ConfigEvents;
+use Drupal\oe_list_pages_open_vocabularies\SearchApiConfigurator;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Create search api configuration when vocabularies associations are edited.
+ */
+class OpenVocabulariesConfigSubscriber implements EventSubscriberInterface {
+
+  /**
+   * The search api configurator.
+   *
+   * @var \Drupal\oe_list_pages_open_vocabularies\SearchApiConfigurator
+   */
+  protected $configurator;
+
+  /**
+   * Constructs an OpenVocabulariesConfigSubscriber object.
+   */
+  public function __construct(SearchApiConfigurator $configurator) {
+    $this->configurator = $configurator;
+  }
+
+  /**
+   * Creates a new field in search index and an associated facet.
+   *
+   * @param \Drupal\Core\Config\ConfigCrudEvent $event
+   *   The Event to process.
+   */
+  public function onSave(ConfigCrudEvent $event) {
+    $config = $event->getConfig();
+    if (strpos($config->getName(), 'open_vocabularies.open_vocabulary_association') !== 0) {
```

Comment:
On second thought, maybe we should use proper entity hooks since the association is an entity no? I see you need the originals so presave is your best bet.